app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/AuthViewModel.kt

```diff
@@ -88,54 +83,34 @@ class AuthViewModel(
         val account = checkNotNull(account)
 
         viewModelScope.launch {
-            val config = findOAuthConfiguration(account)
-            if (config == null) {
-                _uiState.update { AuthFlowState.NotSupported }
-                return@launch
-            }
-
             try {
-                startLogin(account, config)
+                startLogin(account)
             } catch (e: ActivityNotFoundException) {
                 _uiState.update { AuthFlowState.BrowserNotFound }
             }
         }
     }
 
-    private suspend fun startLogin(account: Account, config: OAuthConfiguration) {
+    private suspend fun startLogin(account: Account) {
         val authRequestIntent = withContext(Dispatchers.IO) {
-            createAuthorizationRequestIntent(account.email, config)
+            getOAuthRequestIntent.execute(account.incomingServerSettings.host!!, account.email)
+                .let { result ->
+                    when (result) {
+                        is DomainContract.UseCase.GetOAuthRequestIntent.GetOAuthRequestIntentResult.NotSupported -> {
+                            _uiState.update { AuthFlowState.NotSupported }
```

Comment:
It feels odd to have one side effect here, but handle the other one (starting activity) at the very end of the function.

```suggestion
        val result = withContext(Dispatchers.IO) {
            // Since GetOAuthRequestIntent is suspending, maybe the withContext(Dispatchers.IO) should be moved there?
            getOAuthRequestIntent.execute(account.incomingServerSettings.host!!, account.email)
        }

        when (result) {
            is GetOAuthRequestIntentResult.NotSupported -> {
                _uiState.update { AuthFlowState.NotSupported }
            }
            is GetOAuthRequestIntentResult.Success -> {
                resultObserver.login(result.intent)
            }
        }
```